paywall/src/utils/userMetadata.ts

```diff
@@ -0,0 +1,30 @@
+import { UserMetadata, MetadataInput } from '../unlockTypes'
+
+export function getProtectedInputs(
+  inputs: MetadataInput[]
+): { [name: string]: boolean } {
+  let result: { [key: string]: boolean } = {}
+  inputs.forEach(input => (result[input.name] = input.protected || false))
+  return result
+}
+
+export function formResultToMetadata(
+  formResult: { [key: string]: string },
+  inputs: MetadataInput[]
+): UserMetadata {
+  let result: UserMetadata = {
```

Comment:
Let's make sure we have the same format as used on https://github.com/unlock-protocol/unlock/blob/master/newsletter/utils/token.js